crates/rpc/src/v02/method/add_declare_transaction.rs

```diff
@@ -49,35 +49,41 @@ pub async fn add_declare_transaction(
     context: RpcContext,
     input: AddDeclareTransactionInput,
 ) -> Result<AddDeclareTransactionOutput, AddDeclareTransactionError> {
-    let Transaction::Declare(tx) = input.declare_transaction;
-    let contract_definition: ContractDefinition = tx
-        .contract_class
-        .try_into()
-        .map_err(|e| anyhow::anyhow!("Failed to convert contract definition: {}", e))?;
-
-    let response = context
-        .sequencer
-        .add_declare_transaction(
-            tx.version,
-            tx.max_fee,
-            tx.signature,
-            tx.nonce,
-            contract_definition,
-            tx.sender_address,
-            input.token,
-        )
-        .await?;
-
-    Ok(AddDeclareTransactionOutput {
-        transaction_hash: response.transaction_hash,
-        class_hash: response.class_hash,
-    })
+    match input.declare_transaction {
+        Transaction::Declare(BroadcastedDeclareTransaction::V1(tx)) => {
+            let contract_definition: ContractDefinition = tx
+                .contract_class
+                .try_into()
+                .map_err(|e| anyhow::anyhow!("Failed to convert contract definition: {}", e))?;
+
+            let response = context
+                .sequencer
+                .add_declare_transaction(
+                    tx.version,
+                    tx.max_fee,
+                    tx.signature,
+                    tx.nonce,
+                    contract_definition,
+                    tx.sender_address,
+                    input.token,
+                )
+                .await?;
+
+            Ok(AddDeclareTransactionOutput {
+                transaction_hash: response.transaction_hash,
+                class_hash: response.class_hash,
+            })
+        }
+        Transaction::Declare(BroadcastedDeclareTransaction::V2(_tx)) => todo!("fixme 0.11.0"),
```

Comment:
Looks like we'll need some changes to `AddTransaction` too so that it can represent a v02 declare... That's actually a pre-requirement for call/estimateFee support as well.

Unfortunately we'll probably have to wait with implementing that until we have the 0.11 cairo-lang code...